Add Organization Management Documentation (#1531)
balena-io/docs: pages/learn/manage/organizations.md

---
title: Organization management
---

# Organization management

> An **organization** in balenaCloud is a set of users (members and roles), teams, and applications. Which in turn make collaboration easier and access control more streamlined than ever for your applications and devices. Organizations are available for use in all balenaCloud accounts, blending in with existing features and workflows all the while empowering fleet owners to build bigger, better managed fleets.

Author: @pdcastro We haven't released organizations yet to all users. That's the reason you aren't able to see it at this point from your free account (Whether staging or prod). Alison did mention when organizations will be released the feature will be available to all users and won't be part of paid plans. So that's that. 

I do agree documentation needs to be more low key. That's the reason I added the `What can you do` section below the introduction paragraph where the points are clear with no keywords, no-frills. But, yes I agree I have gone too far with that one. :joy:

> An **organization** in balenaCloud is a set of users (members and roles), teams, and applications. Which in turn make collaboration easier and access control more streamlined than ever for your applications and devices. Organizations are available for use in all balenaCloud accounts, blending in with existing features and workflows all the while empowering fleet owners to build bigger, better managed fleets.

Reviewer: Oh, I missed the new `What can you do` section, sorry. There is some overlap between my suggestion above and the new section, so it needs to be changed a bit.

> An **organization** in balenaCloud is a set of users (members and roles), teams, and applications. Which in turn make collaboration easier and access control more streamlined than ever for your applications and devices. Organizations are available for use in all balenaCloud accounts, blending in with existing features and workflows all the while empowering fleet owners to build bigger, better managed fleets.

Author: @pdcastro How's it looking now? With screenshots too.

## What can you do with organizations?

With organizations you can:

* Invite users to an orgnization as members.

> * Create teams of those members for easier collaboration across applications.

Reviewer: ```
* Invite users to an organization as members, and group them in teams.
* Manage access control at the team level, associating applications with teams.
```

* Bulk provide explicit access to applications to individual members or teams now. 

> * Manage each member, team, or application with easier controls to help meet security and regulatory rules.

Reviewer: _"Manage X with easier controls"_ is not exactly what helps meet security and regulatory rules... What helps meet security and regulatory rules is the ability set access control to apps at a granular level, with user roles like observer / operator / developer, and the ability to change and revoke such roles. I think...

---
title: Organization management
---

# Organization management

> An **organization** in balenaCloud is a group of members, teams, and applications with control over accessiblity and visiblity. Organizations allow granular access control at the application level through multiple roles that can be assigned to individual nembers or teams. This in turns helps to improve the visibility of who has access to which applications making collaboration easier. 

Reviewer: * The words "accessiblity", "visiblity", "nembers" have typos in them - it would be a good idea to run a spell checker over the whole document.
* I'd rephrase _"with control over accessibility and visibility"_ (especially in the first sentence of this document) because "accessibility" generally refers to making computer systems accessible to people with disabilities or impairments to vision, hearing, etc, such as using larger fonts, high contrast, closed captions in video, etc. I think you meant access control instead of accessibility. _"control over visibility"_ leaves one wondering, _"visibility over what?"_ or _"why would I want to control visibility?"..._
* _"a group of members" (especially in the first sentence of this document) may sound puzzling to readers -- members of what, what members? -- not realizing that member means a balenaCloud user.

Suggestion:
```
An **organization** in balenaCloud is a group of users (members), teams, and applications,
with an access control system. Organizations allow the assignment of access control roles
to members or teams, improving visibility and reasoning over who has access to what applications
and, consequently, to device fleets.
```

By the way, in our quest for the soul of organizations, is an organization an access control system? Where "resources" are applications and "actors" are users and teams (which are groups or users). In which case:

```
An _organization_ in balenaCloud is an access control system for balena applications.
Organization _members_ are balenaCloud users who may be grouped in _teams,_ and
assigned _roles_ such as developer or operator. Organizations improve visibility and
reasoning over who has access to what applications and, consequently, to device fleets.

A balenaCloud account may contain one or more organizations, but a balena application
may be added to a single organization.
```

Author: My bad on the typos, I have checked the document before but will run the checker just to be safe. I liked both your suggestions so how about this as the final version. Pushed it just now.

> Organizations are available for use in all balenaCloud accounts, and multiple organizations can be created by a single user. Organizations blend in with existing features and workflows improving upon what's already there.

Reviewer: A sentence like _"Organizations blend in with existing features and workflows improving upon what's already there"_ is not really helpful to readers of this document in my opinion, and could be deleted.

_"multiple organizations can be created by a single user"_ does not make it clear whether such a user is using one or more balenaCloud user accounts. I think the following is a more useful fact to be documented (as always, assuming it is correct...).

```
A balenaCloud account may contain one or more organizations, but a balena application
may be added to a single organization.
```

---
title: Organization management
---

# Organization management

> An **organization** in balenaCloud is a group of users (members), teams, and applications,
> with an access control system. Organization _members_ are balenaCloud users who may be grouped into _teams_, and can be assigned _roles_ such as developer, observer, or operator in balenaCloud applications. Organizations essentially improve visibility and reasoning over who has access to which applications and, consequently, to which device fleets.
> 
> The feature is available for use in all balenaCloud accounts irrespective of plans. A balenaCloud account may contain one or more organizations.

Reviewer 2: ```suggestion
In balenaCloud, users organize around and collaborate in _organizations_. An organization is a group of users, teams and applications, and is managed by one or more organization administrators. An organization can have any number of members that can be grouped into teams and granted varying levels of access to its applications according to their assigned role.

Administrators are implicitly granted the ability to perform any possible action to the organization and its associated resources. This includes inviting and removing members, managing applications, teams and billing.

Each user can create or be a member of an unlimited number of organizations. An organization is automatically created upon user registration, and that user is assigned as its administrator. This organization is managed like any other, but currently cannot be deleted.

Billing applies to each organization separately. Administrators can at any time upgrade an organization to the appropriate billing plan as needed.
```

Sorry, I started trying to improve this blurb a bit but ended up with a bunch of text. Feel free to edit and split it up and distribute in the appropriate sections as you like.

I tried to refine the "organization" definition and highlight admins and billing.

Author: No worries, @dfunckt I will add all these suggestions in the batch and see them as a complete document. If I have any other changes will mention here.

### Inviting members to an organization

With your organization created, let's invite some members. Only the administrator can invite users to an organization. To invite, click the `Manage members` button on the organization summary dashboard. On the page that opens, click the `Invite member` button to open the `Invite organization member` modal. Here, you can either add users with their email address or with their balena username after assigning them roles in your organization. Click `Add member` to invite the user to your organization.

<img alt="Invite Organization Member Modal" src="/img/common/app/invite_org_member.png">

Depending on the role assigned, they will be able to access, see, and manage both apps and other members inside the organization. To change a member's role, navigate to the `Organization Members` tab in the sidebar. Next, change the `Organization role` of the desired user using the dropdown list against their username.

<img alt="Change Organization Member Roles" src="/img/common/app/modify_org_roles.png" width="100%">

## Creating teams in your organization

> With organizations, you can group your existing members to form teams as well. Teams are a robust way to manage members by their function, and bulk provides access to applications as needed. Only administrators are allowed to create a team. To form a team, navigate to the `Organization Teams` tab in the sidebar. On the page that opens, click the `Create Team` button and enter the team's name that is being created.

Reviewer 2: ```suggestion
Organization members can be grouped into teams. Teams provide a way to manage members according to their function. Only administrators can create and delete teams, and manage their members.

To create a team, navigate to the `Organization Teams` page via the sidebar and click the `Create Team` button to enter a name for the team.
```

<img alt="Create Team in Organization" src="/img/common/app/create_team.png" width="100%">

> When the team is created, you can start adding members and applications to that specific team through the `Manage members` and `Manage applications` buttons, respectively.

Reviewer 2: ```suggestion
Once the team is created, add members and grant the team access to applications by clicking on the `Manage members` and `Manage applications` buttons, respectively.
```

<img alt="Organization Team Summary" src="/img/common/app/org_team_summary.png" width="100%">

## Creating new applications under your organization

> With members invited to the organization, let's get started on creating new applications under an organization. Navigate to the `Organization Applications` tab in the sidebar to find the `Create application` button for creating a new application under your organization. Learn more about creating an application from the [getting started](getting-started) guide.

Reviewer 2: I may be misreading the document, but it feels like it can't decide whether its function is to provide reference-level documentation or be a kind of "getting started with orgs" guide. It's perhaps okay as a start, but I personally think we should revisit soon after release.

Author: @dfunckt I started out with a purely referential piece of documentation complementing the blog we are writing but I agree in the end it seems to have a more a leaner, getting started guide tone.

<img alt="New Application under Organization" src="/img/common/app/create_app_under_org.png" width="100%">

### Moving existing applications to an organization

Moving an existing application to an organization is a common task accomplished by [transferring ownership](transferring-ownership) of your application.

<img alt="Transfer Ownership" src="/img/common/app/transfer_ownership.png" width="100%">

> ### Providing access to multiple applications

Reviewer 2: ```suggestion
### Granting members access to applications
```

> With organizations, providing access to individual members or teams becomes a hassle-free process. There are mainly three application-level roles that can be assigned, [`Developer`](developer), [`Observer`](observer), and [`Operator`](operator). Administrators have the [Owner](owner) role by default for all applications under an organization and are the only ones that can provide access to applications.

Reviewer 2: ```suggestion
Organization administrators can grant access to applications either directly for individual members or for teams. There are three application-level roles that can be assigned, [`Developer`](developer), [`Observer`](observer), and [`Operator`](operator), which grant varying levels of access. Administrators are implicitly granted full access to all applications of the organization.
```

> - **Providing access to members** - Once you are done inviting members, click the username of the member you want to provide access to from the `Members` list. When the member's summary page opens, click `Manage applications` to see all applications available under the org. From there, click the `Add to applications` button to add the member to one or more applications available under the org with the desired role.

Reviewer 2: Given the above blurb:

```suggestion
- **Granting access to applications for individual members** - Once you are done inviting members, click the username of the member you want to provide access to from the `Members` list. When the member's summary page opens, click `Manage applications` to see all applications available under the org. From there, click the `Add to applications` button to add the member to one or more applications available under the org with the desired role.
```

### Granting members access to applications

Organization administrators can grant access to applications either directly for individual members or for teams. There are three application-level roles that can be assigned, [`Developer`](developer), [`Observer`](observer), and [`Operator`](operator), which grant varying levels of access. Administrators are implicitly granted full access to all applications of the organization.

> - **Granting access to applications for individual members** - Once you are done inviting members, click the username of the member you want to provide access to from the `Members` list. When the member's summary page opens, click `Manage applications` to see all applications available under the org. From there, click the `Add to applications` button to add the member to one or more applications available under the org with the desired role.

Reviewer 2: Sorry, I think my previous suggestion is not proper English and maybe the following is better:

```suggestion
- **Granting individual members access to applications** - Once you are done inviting members, click the username of the member you want to provide access to from the `Members` list. When the member's summary page opens, click `Manage applications` to see all applications available under the org. From there, click the `Add to applications` button to add the member to one or more applications available under the org with the desired role.
```

> - **Providing access to teams** - Once you are done forming a team, navigate to the team, you want to provide access to in the `Organization Teams` tab.  Next, click the `Manage applications` button on the team page to see all applications available under the org. From there, click the `Add applications to team` button to add the team to one or more applications available under the org with the desired roles. All members of that team will be provided access to the selected application(s) with the specified role per application.

Reviewer 2: ```suggestion
- **Granting access to applications for teams** - Teams and their members can be granted access to specific applications from the `Organization Teams` tab.  Click the `Manage applications` button on the team page to see all applications available under the org. From there, click the `Add applications to team` button to add the team to one or more applications available under the org with the desired roles. All members of that team will be provided access to the selected application(s) with the specified role per application.
```

<img alt="Provide Member Access to Applications" src="/img/common/app/application_access_to_member.png" width="100%">

> - **Granting access to applications for teams** - Teams and their members can be granted access to specific applications from the `Organization Teams` tab.  Click the `Manage applications` button on the team page to see all applications available under the org. From there, click the `Add applications to team` button to add the team to one or more applications available under the org with the desired roles. All members of that team will be provided access to the selected application(s) with the specified role per application.

Reviewer 2: Same here:

```suggestion
- **Granting teams access to applications** - Teams and their members can be granted access to specific applications from the `Organization Teams` tab.  Click the `Manage applications` button on the team page to see all applications available under the org. From there, click the `Add applications to team` button to add the team to one or more applications available under the org with the desired roles. All members of that team will be provided access to the selected application(s) with the specified role per application.
```

> ## Users as an organization - hassle-free app sharing
> 
> With organizations being a crucial feature for collaboration, we wanted the same for each individual user too. Hence, every user in balenaCloud owns an organization by default named after their username. In this **default organization**, all applications and devices are private until **explicitly shared** by the user.
> 
> Default organizations can be used to develop & test applications. When ready, these applications can be shared with other members or teams by inviting users to your organization. A default organization is no different from any other organization that you can create. This way, each user is empowered to collaborate, manage, and work on their fleets securely.

Reviewer 2: 😁

```suggestion
```

---
title: Organization management
---

# Organization management

> Organizations in balenaCloud are based around the idea of easier collaboration and robust management of members and applications. The feature blends in with current workflows all the while improving what is already there.

Reviewer 3: Agree with Paulo on above. We want a bit more info on what they are and what they will be used for up front

> Organizations in balenaCloud are based around the idea of easier collaboration and robust management of members and applications. The feature blends in with current workflows all the while improving what is already there.

Author: I agree @pdcastro the introduction paragraph was something I was holding off for later in the view of plucking a good definition out of the blog John is writing. But, yes this is needed and extremely important for folks to read when they land on the page. I will keep this open, and post what we are adding for a definition.

> Organizations in balenaCloud are based around the idea of easier collaboration and robust management of members and applications. The feature blends in with current workflows all the while improving what is already there.

Reviewer 4: I like the core of what Paulo wrote, but we should add more of the "Why" to the description. For example, "orgs make it easier for teams of developers to collaborate across one or many applications", "orgs allow you to create better distinctions between development and production projects)", "orgs allow more specific access and views of project and codes to help meet security and regulatory rules", etc.

> Organizations in balenaCloud are based around the idea of easier collaboration and robust management of members and applications. The feature blends in with current workflows all the while improving what is already there.

Author: @jtonello  You got it, should we also link back to the blog post. Since that covers a lot of the why and other questions that might pop in people's head when going through the feature.

> Organizations in balenaCloud are based around the idea of easier collaboration and robust management of members and applications. The feature blends in with current workflows all the while improving what is already there.

Author: ![Screenshot from 2020-12-04 13-13-35](https://user-images.githubusercontent.com/22801822/101135926-8bffe480-3632-11eb-82bd-2caa9f086237.png)

How's this looking @pdcastro @shaunmulligan @jtonello Will push this as well.

> ## Sign up

Reviewer 3: Maybe this feels a bit too obvious? I would probably just remove it.

Author: Roger that, added it because I was following the same template as account management document

{{> "general/signUp"}}

## Creating a new organization

To create a new organization, navigate to your balenaCloud dashboard and click the `Organizations` tab on the sidebar.

<img alt="Organizations tab" src="/img/common/app/organizations_tab.png" width="100%">

Next, click the `Create Organization` button to get the create organization modal below. Fill in the name of your organization and you are all set. This is also the page where you can check all organizations that you have access to. The user creating an organization is default given administrator access to that organization. Only administrators can delete an organization. Alternatively, anyone can leave an organization by clicking the `Leave organization` button on the organization summary page.

> 

Reviewer: `is default` → `is by default`

* **Member**: Members are provided with read-only access to the organization with explicit access to its applications and teams. Roles on the application level are assigned by administrators. Members have no visibility over other applications and teams which they haven't been given access to in the organization.

### Inviting members to an organization

Now, with your organization created let's invite some members. Only the administrator can invite users to an organization. To invite, click the `Manage members` button on the organization summary dashboard. On the page that opens, click the `Invite member` button to open the `Invite organization member` modal. Here, you can either add users with their email or with their balena username. After assigning them roles in your organization. Click `Add member` to invite the user to your organization.

<img alt="Invite Organization Member Modal" src="/img/common/app/" width="100%">

> On basis of the role assigned, they will be able to access, see, and manage both apps and other members inside the organization. To change the roles of a member, navigate to the `Organization Members` tab in the sidebar and change the `Organization role` of the desired user by using the dropdown list against their username.

Reviewer: `On basis of` → `On the basis of`
_or_ `Depending on the role assigned,`

<img alt="Change Organization Member Roles" src="/img/common/app/" width="100%">

## Creating teams in your organization

> With organizations, you can group your existing members to form teams as well. Teams are a robust way to manage members per their function and bulk provide access to applications as needed. Only administrators are allowed to create a team. To form a team, navigate to `Organization Teams` in the sidebar of the organization summary page. On the page that opens, click the `Create Team` button and enter the name of the team that is being created.

Reviewer: * `per` → `by`
* `and bulk provide access` → `and to provide access in bulk`

<img alt="Create Team in Organization" src="/img/common/app/" width="100%">

When the team is created, you can start adding members and applications to that specific team through the `Manage members` and `Manage applications` buttons respectively.

<img alt="Organization Team Summary" src="/img/common/app/" width="100%">

## Creating new applications under your organization

With members invited to the organization, let's get started on creating new applications under an organization. Navigate to the `Organization Applications` tab in the sidebar to find the `Create application` button for creating a new application under your organization. Learn more about creating an application from the [getting started](getting-started) guide.

<img alt="New Application under Organization" src="/img/common/app/" width="100%">

### Moving existing applications to an organization

> Moving an existing application from your personal or any other organization to another organization is a common task that can be accomplished by [transferring ownership](transferring-ownership) of your application.

Reviewer: `from your personal or any other organization to another organization`
→ `to an organization`
I.e.,
`Moving an existing application to an organization is a common task ...`

<img alt="Transfer Ownership" src="/img/common/app/" width="100%">

### Providing access to multiple applications

> With organizations, providing access to individual members or teams becomes a hassle-free process. There are mainly 3 application-level roles that can be assigned, [`Developer`](developer), [`Observer`](observer) or [`Operator`](operator). Administrator default to [`Owner`](owner) role to all applications under a organization and are the only ones that can provide access to applications. Administrators won't be listed in members page of an organization.  

Reviewer: * `or Operator` → `and Operator`
* `Administrator default to [`Owner`](owner) role to all applications`
  → `Administrators have the [`Owner`](owner) role by default for all applications`

> - **Providing access to members** - Once you are done inviting members, click the username you want to provide access to from the `Members` list. When the member page opens, click `Manage applications` to see all applications available under the org. From there, click the `Add applications to member` button to add the member to one or more applications available under the org with the desired role that you want to assign.

Reviewer: * `member page` → `member's page`
* delete `that you want to assign`

<img alt="Provide Member Access to Applications" src="/img/common/app/" width="100%">

> - **Providing access to teams** - Once you are done creating a team, navigate to the team you want to provide access in the `Organization Teams` tab.  Next, click the `Manage applications` button on the team page to see all applications available under the org. From there, click the `Add applications to team` button to add the team to one or more applications available under the org with the desired roles that you want to assign. All members of that team will be provided access to the selected application with the specified role per application.

Reviewer: delete `that you want to assign`

<img alt="Provide Team Access to Applications" src="/img/common/app/" width="100%">

> ## Users as an organization - Sharing your apps just became hassle-free

Reviewer: `## Users as an organization - hassle-free app sharing`

> With organizations being a crucial feature for collaboration, we wanted the same for individual users. Each user in balenaCloud has their own "personal" organization named after their username. This "personal organization" with all applications and devices under it are private until explicitly shared by the user.

Reviewer 3: I believe we don't want to call it "personal" but rather "default" org.

Author: Yes, that's a discussion-worthy thing we need to take a decision on the blog as well, I will check in on that before making a decision.

Author: @shaunmulligan We will be going with "default" org everywhere.

> Personal organizations can be used to develop & test applications. When ready, these applications can be shared with other members or teams by inviting users to your organization. A personal organization is no different from any other organization that you can create and works in the same way. With this, each user is empowered to collaborate, manage and work on their fleets more easily & securely.

Reviewer: `With this,` → `This way,`
